src/averageAges.js

```diff
@@ -55,7 +81,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredTab = (onlyWithSon)
+    ? people.filter(filterMan).filter(filterChilds)
+    : people.filter(filterChilds);
+
+  function filterMan(person) {
+    return person.sex === 'm';
+  }
+
+  function filterChilds(person) {
+    return people.some(el => el.name === person.mother);
+  };
+
+  return filteredTab.reduce((prev, next) => {
+    return prev + (next.born - people.find(el => el.name === next.mother).born);
+  }, 0) / filteredTab.length;
```

Comment:
Try to make this code more readible by introducing some better naming and code formating